CMakeLists.txt

```diff
@@ -0,0 +1,125 @@
+# Copyright 2010-2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License").
+# You may not use this file except in compliance with the License.
+# A copy of the License is located at
+#
+#  http://aws.amazon.com/apache2.0
+#
+# or in the "license" file accompanying this file. This file is distributed
+# on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+# express or implied. See the License for the specific language governing
+# permissions and limitations under the License.
+
+cmake_minimum_required(VERSION 3.0)
+project(aws-c-common)
+
+set(PROJECT_NAME aws-c-common)
+
+file(GLOB AWS_COMMON_HEADERS
+        "include/aws/common/*.h"
+        )
+
+file(GLOB AWS_COMMON_PRIV_HEADERS
+        "include/aws/common/private/*.h"
+        )
+
+file(GLOB AWS_COMMON_SRC
+        "source/*.c"
+        )
+
+if (WIN32)
+    file(GLOB AWS_COMMON_OS_SRC
+            "source/windows/*.c"
+            )
+
+    if (MSVC)
+        source_group("Header Files\\aws\\common" FILES ${AWS_COMMON_HEADERS})
+        source_group("Header Files\\aws\\common\\private" FILES ${AWS_COMMON_PRIV_HEADERS})
+        source_group("Source Files" FILES ${AWS_COMMON_SRC})
+        source_group("Source Files\\windows" FILES ${AWS_COMMON_OS_SRC})
+    endif ()
+
+    set(PLATFORM_LIBS "Kernel32")
+else ()
+    file(GLOB AWS_COMMON_OS_SRC
+            "source/posix/*.c"
+            )
+
+    set(PLATFORM_LIBS "pthread" "rt")
+endif ()
+
+file(GLOB COMMON_HEADERS
+        ${AWS_COMMON_HEADERS}
+        ${AWS_COMMON_PRIV_HEADERS}
+        )
+
+file(GLOB COMMON_SRC
+        ${AWS_COMMON_SRC}
+        ${AWS_COMMON_OS_SRC}
+        )
+
+if (BUILD_SHARED_LIBS)
+    set(LIBTYPE SHARED)
+    if (WIN32)
+        add_definitions("-DUSE_IMPORT_EXPORT")
+    endif ()
+else ()
+    set(LIBTYPE STATIC)
+endif ()
+
+add_library(aws-c-common ${LIBTYPE} ${COMMON_HEADERS} ${COMMON_SRC})
```

Comment:
${LIBTYPE} is not needed. CMake builds the correct thing based on the value of BUILD_SHARED_LIBS